packages/csvviewer/src/widget.ts

```diff
@@ -122,40 +134,90 @@ export class GridSearchService {
       );
     };
 
-    for (; this._row < model.rowCount('body'); this._row++) {
-      for (; this._column < model.columnCount('body'); this._column++) {
-        const cellData = model.data('body', this._row, this._column) as string;
-        if (cellData.indexOf(searchText) !== -1) {
+    const increment = reverse ? -1 : 1;
+    this._column += increment;
+    for (
```

Comment:
Good question. I left the searching logic basically alone, so it probably performs on par with what it was before (with the exception that now I do a regex match instead of a string lookup).